In [2]:
import instaloader
from tqdm.notebook import tqdm
import pandas as pd
from parsing.insta_parse import get_info, get_user_names

## A few words about using the Instaloader

**The library provides the ability to collect information from Instagram accounts both through another existing account and without logging to Instagram. The first method allows you to read more information (info about highlights, followers and followees), but it's unstable due to the restrictive Istagram policy. Therefore, we will collect the database of usernames in this way through a fake account, and parse the data itself from the collected accounts - without logging into Instargam.**

***The highlights data was not specifically collected due to the long-term nature of their reading and to speed up the process of predicting the future model**

**I selected 6 accounts whose subscriptions were diverse in their properties.**

Библиотека Instaloader предоставляет возможность собирать информацию с Instagram-аккаунтов как через другой существующий аккаунт, так и без входа в Instagram. Первый способ позволяет считывать больше информации (данные об актуальных, подписчиках и подписках), однако он нестабилен в виду ограничительной политики Istagram. Поэтому собирать базу username-ов будем данным способом через fake-овый аккаунт, а парсить сами данные с собранных аккаунтах - без  осуществления входа в Instargam.

*Данные об актуальных умышленно не были спарсены из-за долговременности их считывания и для ускорения процесса предсказания будущей модели

Мною было отобрано 6 аккаунтов, подписки которых были по своим свойствам разнообразны.

In [3]:
main_accs = ['owmina', 'polsorok', 'androsovanst', 'kkarina_shults', 'veronika_dobra', 'brows_by.a']

In [5]:
L = instaloader.Instaloader() # initializing the Instaloader class without logging in to the account

L.load_session_from_file('brows_by.a') # log in

Loaded session from C:\Users\1\AppData\Local\Instaloader\session-brows_by.a.


In [6]:
# collect user names
df = pd.DataFrame(columns=['user_name'])
for acc in tqdm(main_accs):
    user_names = get_user_names(L, acc, num_accounts=1000, from_followers=False)
    df = pd.concat([df, user_names])

801it [00:32, 24.86it/s]:00<?, ?it/s]
240it [00:10, 22.95it/s]:36<03:01, 36.31s/it]
415it [00:21, 19.47it/s]:50<01:34, 23.58s/it]
928it [00:32, 28.24it/s]:16<01:13, 24.45s/it]
313it [00:10, 28.98it/s]:52<00:57, 28.85s/it]
116it [00:03, 31.43it/s]:08<00:24, 24.25s/it]
100%|██████████| 6/6 [02:20<00:00, 23.37s/it]


In [10]:
df.reset_index(drop=True, inplace=True)

In [11]:
L = instaloader.Instaloader() # initializing the Instaloader class without logging in to the account

for i, acc in enumerate(tqdm(df['user_name'])):
    try:
        info = get_info(L, acc, num_posts=10) # filling in the DataFrame
    except Exception:
        continue

    for feat in info:
        df.loc[i, feat] = info[feat]

  0%|          | 9/2813 [00:24<2:43:50,  3.51s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


  1%|          | 16/2813 [00:46<2:47:40,  3.60s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


  3%|▎         | 75/2813 [03:36<2:11:47,  2.89s/it]


Too many queries in the last time. Need to wait 447 seconds, until 13:40.


JSON Query to nasttturs/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
  3%|▎         | 86/2813 [11:38<5:15:07,  6.93s/it]   


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


  5%|▌         | 149/2813 [15:05<3:10:32,  4.29s/it]


Too many queries in the last time. Need to wait 426 seconds, until 13:51.


JSON Query to badluckboii/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
  8%|▊         | 222/2813 [26:28<2:24:20,  3.34s/it]   


Too many queries in the last time. Need to wait 440 seconds, until 14:03.


JSON Query to marina_chuvalskaya/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 10%|█         | 286/2813 [37:13<1:54:18,  2.71s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 11%|█         | 296/2813 [37:39<1:57:29,  2.80s/it]


Too many queries in the last time. Need to wait 436 seconds, until 14:14.


JSON Query to pankartov_media/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 13%|█▎        | 358/2813 [48:18<1:41:23,  2.48s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 13%|█▎        | 370/2813 [48:46<1:21:44,  2.01s/it]


Too many queries in the last time. Need to wait 436 seconds, until 14:25.


JSON Query to yugentear/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 14%|█▍        | 404/2813 [58:01<2:02:43,  3.06s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 16%|█▌        | 444/2813 [1:00:00<1:22:41,  2.09s/it]


Too many queries in the last time. Need to wait 427 seconds, until 14:36.


JSON Query to nochm_lera/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 17%|█▋        | 474/2813 [1:08:57<2:22:45,  3.66s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 18%|█▊        | 493/2813 [1:10:02<1:22:44,  2.14s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 18%|█▊        | 495/2813 [1:10:07<1:30:11,  2.33s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 18%|█▊        | 513/2813 [1:10:59<1:46:44,  2.78s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 18%|█▊        | 515/2813 [1:11:03<1:34:09,  2.46s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 18%|█▊        | 518/2813 [1:11:15<2:03:53,  3.24s/it]


Too many queries in the last time. Need to wait 416 seconds, until 14:47.


JSON Query to flacon_magazine/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 19%|█▊        | 526/2813 [1:18:47<8:58:55, 14.14s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 19%|█▉        | 535/2813 [1:19:18<2:25:01,  3.82s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 20%|█▉        | 553/2813 [1:25:20<21:46:24, 34.68s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 20%|█▉        | 558/2813 [1:25:28<4:45:28,  7.60s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 20%|█▉        | 559/2813 [1:25:34<4:18:00,  6.87s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 20%|██        | 571/2813 [1:26:17<1:56:57,  3.13s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 21%|██        | 591/2813 [1:27:07<1:09:54,  1.89s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query

Too many queries in the last time. Need to wait 128 seconds, until 14:58.


 21%|██▏       | 598/2813 [1:29:47<5:43:23,  9.30s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 21%|██▏       | 602/2813 [1:30:00<3:19:24,  5.41s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 22%|██▏       | 624/2813 [1:31:10<1:42:35,  2.81s/it]


Too many queries in the last time. Need to wait 300 seconds, until 15:05.


JSON Query to _willie_white_/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 24%|██▎       | 665/2813 [1:38:17<1:48:47,  3.04s/it] 


Too many queries in the last time. Need to wait 127 seconds, until 15:09.


 25%|██▍       | 695/2813 [1:42:16<3:09:24,  5.37s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 25%|██▍       | 697/2813 [1:42:21<2:15:03,  3.83s/it]


Too many queries in the last time. Need to wait 296 seconds, until 15:16.


JSON Query to oleeex2/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 25%|██▍       | 698/2813 [1:47:23<54:44:10, 93.17s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 26%|██▌       | 729/2813 [1:49:01<2:28:49,  4.28s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 26%|██▌       | 736/2813 [1:49:18<1:59:44,  3.46s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 26%|██▌       | 738/2813 [1:49:22<1:37:22,  2.82s/it]


Too many queries in the last time. Need to wait 124 seconds, until 15:21.


 27%|██▋       | 753/2813 [1:52:29<3:00:32,  5.26s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 27%|██▋       | 758/2813 [1:52:50<3:07:13,  5.47s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 27%|██▋       | 771/2813 [1:53:34<2:04:05,  3.65s/it]


Too many queries in the last time. Need to wait 291 seconds, until 15:27.


JSON Query to roseman924/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 28%|██▊       | 776/2813 [1:58:36<13:07:09, 23.19s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 28%|██▊       | 779/2813 [1:58:44<5:22:23,  9.51s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 28%|██▊       | 781/2813 [1:58:51<3:43:52,  6.61s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 28%|██▊       | 791/2813 [1:59:30<1:47:44,  3.20s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 29%|██▉       | 812/2813 [2:00:33<1:50:42,  3.32s/it]


Too many queries in the last time. Need to wait 123 seconds, until 15:32.


 30%|███       | 845/2813 [2:04:46<2:11:54,  4.02s/it] 


Too many queries in the last time. Need to wait 283 seconds, until 15:39.


JSON Query to sedakasparova/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 31%|███       | 864/2813 [2:10:38<2:23:28,  4.42s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 31%|███▏      | 886/2813 [2:11:36<1:17:50,  2.42s/it]


Too many queries in the last time. Need to wait 123 seconds, until 15:43.


 33%|███▎      | 915/2813 [2:15:38<2:30:00,  4.74s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 33%|███▎      | 919/2813 [2:15:54<2:31:25,  4.80s/it]


Too many queries in the last time. Need to wait 283 seconds, until 15:50.


JSON Query to alexandra.geyfman/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 34%|███▍      | 960/2813 [2:22:56<1:16:46,  2.49s/it] 


Too many queries in the last time. Need to wait 114 seconds, until 15:54.


 35%|███▌      | 993/2813 [2:26:59<2:12:48,  4.38s/it] 


Too many queries in the last time. Need to wait 284 seconds, until 16:01.


JSON Query to vorona2002/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 35%|███▌      | 998/2813 [2:31:59<12:03:16, 23.91s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 36%|███▌      | 1018/2813 [2:33:08<1:38:11,  3.28s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 37%|███▋      | 1034/2813 [2:34:01<1:28:11,  2.97s/it]


Too many queries in the last time. Need to wait 115 seconds, until 16:05.


 37%|███▋      | 1054/2813 [2:36:56<1:07:33,  2.30s/it] 


Too many queries in the last time. Need to wait 20 seconds, until 16:06.


 38%|███▊      | 1067/2813 [2:38:13<2:19:15,  4.79s/it]


Too many queries in the last time. Need to wait 272 seconds, until 16:12.


JSON Query to elianrussia/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 39%|███▉      | 1108/2813 [2:45:00<56:30,  1.99s/it]   


Too many queries in the last time. Need to wait 121 seconds, until 16:16.


 40%|████      | 1128/2813 [2:48:01<1:07:40,  2.41s/it] 


Too many queries in the last time. Need to wait 19 seconds, until 16:17.


 40%|████      | 1132/2813 [2:48:45<3:40:04,  7.86s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 41%|████      | 1141/2813 [2:49:15<1:41:57,  3.66s/it]


Too many queries in the last time. Need to wait 277 seconds, until 16:23.


JSON Query to miglioteba/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 42%|████▏     | 1182/2813 [2:56:24<1:23:45,  3.08s/it] 


Too many queries in the last time. Need to wait 100 seconds, until 16:27.


 43%|████▎     | 1202/2813 [2:59:12<1:18:35,  2.93s/it] 


Too many queries in the last time. Need to wait 15 seconds, until 16:28.


 43%|████▎     | 1203/2813 [2:59:31<3:30:16,  7.84s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 43%|████▎     | 1215/2813 [3:00:20<2:09:24,  4.86s/it]


Too many queries in the last time. Need to wait 279 seconds, until 16:34.


JSON Query to dmtrhnn/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 44%|████▍     | 1233/2813 [3:06:14<1:52:57,  4.29s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 44%|████▍     | 1244/2813 [3:06:45<1:40:30,  3.84s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 44%|████▍     | 1248/2813 [3:07:04<2:30:18,  5.76s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 45%|████▍     | 1256/2813 [3:07:30<1:23:21,  3.21s/it]


Too many queries in the last time. Need to wait 103 seconds, until 16:38.


 45%|████▍     | 1259/2813 [3:09:29<8:24:59, 19.50s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 45%|████▌     | 1276/2813 [3:10:12<52:03,  2.03s/it]  


Too many queries in the last time. Need to wait 20 seconds, until 16:40.


 45%|████▌     | 1279/2813 [3:10:48<3:15:57,  7.66s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 46%|████▌     | 1289/2813 [3:11:31<1:55:06,  4.53s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query

Too many queries in the last time. Need to wait 275 seconds, until 16:45.


JSON Query to fin_tema/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 46%|████▌     | 1299/2813 [3:16:51<3:26:35,  8.19s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 47%|████▋     | 1330/2813 [3:18:41<1:50:21,  4.47s/it]


Too many queries in the last time. Need to wait 97 seconds, until 16:49.


 48%|████▊     | 1350/2813 [3:21:16<1:04:40,  2.65s/it] 


Too many queries in the last time. Need to wait 23 seconds, until 16:51.


 48%|████▊     | 1363/2813 [3:22:32<1:42:55,  4.26s/it]


Too many queries in the last time. Need to wait 281 seconds, until 16:56.


JSON Query to n1chss/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 49%|████▊     | 1371/2813 [3:27:50<5:19:35, 13.30s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 49%|████▉     | 1391/2813 [3:28:59<1:24:32,  3.57s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 50%|████▉     | 1404/2813 [3:29:50<2:06:19,  5.38s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query

Too many queries in the last time. Need to wait 96 seconds, until 17:00.


 50%|████▉     | 1406/2813 [3:31:38<10:18:01, 26.35s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 51%|█████     | 1424/2813 [3:32:21<42:01,  1.82s/it]   


Too many queries in the last time. Need to wait 21 seconds, until 17:02.


 51%|█████     | 1428/2813 [3:33:04<2:44:03,  7.11s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 51%|█████     | 1437/2813 [3:33:33<1:07:28,  2.94s/it]


Too many queries in the last time. Need to wait 284 seconds, until 17:07.


JSON Query to kristinamaksimovna_l/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 51%|█████     | 1441/2813 [3:38:31<12:19:49, 32.35s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 52%|█████▏    | 1456/2813 [3:39:36<1:27:43,  3.88s/it] 


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 53%|█████▎    | 1478/2813 [3:40:54<2:03:11,  5.54s/it]


Too many queries in the last time. Need to wait 88 seconds, until 17:12.


 54%|█████▎    | 1511/2813 [3:44:41<1:10:37,  3.25s/it] 


Too many queries in the last time. Need to wait 284 seconds, until 17:18.


JSON Query to sochi_cases/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 54%|█████▍    | 1525/2813 [3:50:28<2:28:05,  6.90s/it] JSON Query to beautymaniaband_sochi/feed/: 404 Not Found [retrying; skip with ^C]
JSON Query to beautymaniaband_sochi/feed/: 404 Not Found [retrying; skip with ^C]
 55%|█████▌    | 1549/2813 [3:52:01<1:54:50,  5.45s/it]


Too many queries in the last time. Need to wait 96 seconds, until 17:23.


 56%|█████▌    | 1582/2813 [3:55:47<1:09:30,  3.39s/it] 


Too many queries in the last time. Need to wait 284 seconds, until 17:30.


JSON Query to your_english.tutor/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 58%|█████▊    | 1623/2813 [4:03:07<1:34:00,  4.74s/it] 


Too many queries in the last time. Need to wait 96 seconds, until 17:34.


 59%|█████▉    | 1656/2813 [4:06:52<1:01:39,  3.20s/it] 


Too many queries in the last time. Need to wait 284 seconds, until 17:41.


JSON Query to handmade_hobby_/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 60%|██████    | 1697/2813 [4:14:17<43:44,  2.35s/it]   


Too many queries in the last time. Need to wait 91 seconds, until 17:45.


 62%|██████▏   | 1730/2813 [4:18:06<1:15:45,  4.20s/it]


Too many queries in the last time. Need to wait 272 seconds, until 17:52.


JSON Query to svetlana171974_magiya_gadanue/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 63%|██████▎   | 1771/2813 [4:25:26<42:35,  2.45s/it]   


Too many queries in the last time. Need to wait 86 seconds, until 17:56.


 64%|██████▍   | 1804/2813 [4:29:11<1:13:16,  4.36s/it]


Too many queries in the last time. Need to wait 273 seconds, until 18:03.


JSON Query to baranenkova.smm/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 66%|██████▌   | 1845/2813 [4:36:44<48:20,  3.00s/it]   


Too many queries in the last time. Need to wait 78 seconds, until 18:07.


 67%|██████▋   | 1878/2813 [4:40:18<1:22:28,  5.29s/it]


Too many queries in the last time. Need to wait 276 seconds, until 18:14.


JSON Query to elite_kraska_ru/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 68%|██████▊   | 1919/2813 [4:47:49<42:37,  2.86s/it]   


Too many queries in the last time. Need to wait 79 seconds, until 18:18.


 69%|██████▉   | 1944/2813 [4:50:37<39:02,  2.70s/it]  


Too many queries in the last time. Need to wait 15 seconds, until 18:20.


 69%|██████▉   | 1952/2813 [4:51:25<1:32:40,  6.46s/it]


Too many queries in the last time. Need to wait 276 seconds, until 18:25.


JSON Query to postelnoelux_moskva/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 70%|███████   | 1973/2813 [4:57:33<1:04:28,  4.61s/it] 


Too many queries in the last time. Need to wait 16 seconds, until 18:27.


 71%|███████   | 1993/2813 [4:59:01<39:44,  2.91s/it]  


Too many queries in the last time. Need to wait 72 seconds, until 18:29.


 72%|███████▏  | 2018/2813 [5:02:02<58:00,  4.38s/it]  


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 72%|███████▏  | 2025/2813 [5:02:30<1:14:40,  5.69s/it]


Too many queries in the last time. Need to wait 273 seconds, until 18:36.


JSON Query to quest.ono/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 73%|███████▎  | 2046/2813 [5:08:38<57:09,  4.47s/it]   


Too many queries in the last time. Need to wait 18 seconds, until 18:38.


 73%|███████▎  | 2055/2813 [5:09:27<44:47,  3.55s/it]  


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 73%|███████▎  | 2065/2813 [5:10:01<24:32,  1.97s/it]  


Too many queries in the last time. Need to wait 77 seconds, until 18:40.


 75%|███████▍  | 2098/2813 [5:13:34<25:59,  2.18s/it]  


Too many queries in the last time. Need to wait 277 seconds, until 18:47.


JSON Query to sakura_place_/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 75%|███████▍  | 2100/2813 [5:18:18<12:04:07, 60.94s/it]


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 75%|███████▍  | 2105/2813 [5:18:38<2:38:41, 13.45s/it] 


Too many queries in the last time. Need to wait 16 seconds, until 18:48.


 75%|███████▌  | 2118/2813 [5:19:45<56:38,  4.89s/it]  


Too many queries in the last time. Need to wait 16 seconds, until 18:49.


 76%|███████▌  | 2138/2813 [5:21:12<31:21,  2.79s/it]  


Too many queries in the last time. Need to wait 74 seconds, until 18:51.


 77%|███████▋  | 2171/2813 [5:24:49<47:09,  4.41s/it]  


Too many queries in the last time. Need to wait 270 seconds, until 18:58.


JSON Query to marytrufel_krasnodar/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 77%|███████▋  | 2179/2813 [5:29:43<1:43:15,  9.77s/it] 


Too many queries in the last time. Need to wait 16 seconds, until 18:59.


 78%|███████▊  | 2192/2813 [5:30:46<41:31,  4.01s/it]  


Too many queries in the last time. Need to wait 23 seconds, until 19:00.


 79%|███████▊  | 2212/2813 [5:32:30<34:49,  3.48s/it]  


Too many queries in the last time. Need to wait 62 seconds, until 19:03.


 80%|███████▉  | 2245/2813 [5:35:56<42:49,  4.52s/it]  


Too many queries in the last time. Need to wait 268 seconds, until 19:09.


JSON Query to dr.mumish/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 81%|████████  | 2266/2813 [5:41:55<37:21,  4.10s/it]   


Too many queries in the last time. Need to wait 18 seconds, until 19:11.


 81%|████████▏ | 2286/2813 [5:43:37<31:32,  3.59s/it]  


Too many queries in the last time. Need to wait 61 seconds, until 19:14.


 82%|████████▏ | 2309/2813 [5:46:13<25:38,  3.05s/it]  


Too many queries in the last time. Need to wait 16 seconds, until 19:15.


 82%|████████▏ | 2311/2813 [5:46:36<56:23,  6.74s/it]  


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 82%|████████▏ | 2318/2813 [5:47:00<38:37,  4.68s/it]


Too many queries in the last time. Need to wait 270 seconds, until 19:21.


JSON Query to daleart_studio/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 83%|████████▎ | 2326/2813 [5:51:54<1:17:59,  9.61s/it] 


Too many queries in the last time. Need to wait 18 seconds, until 19:21.


 83%|████████▎ | 2339/2813 [5:52:56<27:01,  3.42s/it]  


Too many queries in the last time. Need to wait 24 seconds, until 19:22.


 84%|████████▍ | 2359/2813 [5:54:43<32:23,  4.28s/it]  


Too many queries in the last time. Need to wait 60 seconds, until 19:25.


 84%|████████▍ | 2370/2813 [5:56:36<50:36,  6.85s/it]  


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 85%|████████▍ | 2381/2813 [5:57:16<22:14,  3.09s/it]


Too many queries in the last time. Need to wait 17 seconds, until 19:27.


 85%|████████▍ | 2384/2813 [5:57:42<41:30,  5.80s/it]


HTTP redirect from https://www.instagram.com/web/search/topsearch/ to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/


 85%|████████▍ | 2389/2813 [5:58:05<39:24,  5.58s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 85%|████████▍ | 2390/2813 [5:58:08<33:40,  4.78s/it]


Too many queries in the last time. Need to wait 268 seconds, until 19:32.


JSON Query to poniroma/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 85%|████████▌ | 2398/2813 [6:02:59<1:04:10,  9.28s/it]


Too many queries in the last time. Need to wait 16 seconds, until 19:32.


 86%|████████▌ | 2411/2813 [6:03:58<21:56,  3.27s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query

Too many queries in the last time. Need to wait 25 seconds, until 19:33.


 86%|████████▋ | 2431/2813 [6:05:47<21:23,  3.36s/it]  


Too many queries in the last time. Need to wait 62 seconds, until 19:36.


 87%|████████▋ | 2454/2813 [6:08:20<17:39,  2.95s/it]  


Too many queries in the last time. Need to wait 17 seconds, until 19:38.


 88%|████████▊ | 2464/2813 [6:09:12<27:20,  4.70s/it]


Too many queries in the last time. Need to wait 271 seconds, until 19:43.


JSON Query to _holy___boy_/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 88%|████████▊ | 2472/2813 [6:14:08<57:53, 10.19s/it]  


Too many queries in the last time. Need to wait 15 seconds, until 19:43.


 88%|████████▊ | 2485/2813 [6:15:02<13:57,  2.55s/it]  


Too many queries in the last time. Need to wait 30 seconds, until 19:45.


 89%|████████▉ | 2502/2813 [6:16:49<33:15,  6.42s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 89%|████████▉ | 2505/2813 [6:16:57<19:57,  3.89s/it]


Too many queries in the last time. Need to wait 56 seconds, until 19:47.


 90%|█████████ | 2538/2813 [6:20:17<15:02,  3.28s/it]  


Too many queries in the last time. Need to wait 271 seconds, until 19:54.


JSON Query to sweet_dreams_showroom1/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 91%|█████████ | 2546/2813 [6:25:13<45:10, 10.15s/it]  


Too many queries in the last time. Need to wait 16 seconds, until 19:55.


 91%|█████████ | 2558/2813 [6:26:17<23:42,  5.58s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 91%|█████████ | 2559/2813 [6:26:19<19:13,  4.54s/it]


Too many queries in the last time. Need to wait 18 seconds, until 19:56.


 91%|█████████ | 2562/2813 [6:26:44<23:31,  5.63s/it]


Too many queries in the last time. Need to wait 16 seconds, until 19:56.


 92%|█████████▏| 2579/2813 [6:27:59<14:01,  3.60s/it]


Too many queries in the last time. Need to wait 62 seconds, until 19:58.


 92%|█████████▏| 2596/2813 [6:30:19<15:58,  4.42s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 93%|█████████▎| 2612/2813 [6:31:21<12:05,  3.61s/it]


Too many queries in the last time. Need to wait 273 seconds, until 20:05.


JSON Query to zimakova_oks/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 93%|█████████▎| 2619/2813 [6:36:24<44:31, 13.77s/it]  


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 94%|█████████▎| 2636/2813 [6:37:50<16:05,  5.46s/it]


Too many queries in the last time. Need to wait 16 seconds, until 20:07.


 94%|█████████▍| 2653/2813 [6:39:05<09:54,  3.72s/it]


Too many queries in the last time. Need to wait 59 seconds, until 20:09.


 95%|█████████▍| 2661/2813 [6:40:40<13:12,  5.21s/it]


HTTP redirect from https://www.instagram.com/graphql/query to https://www.instagram.com/accounts/login/?next=/graphql/query


 95%|█████████▌| 2686/2813 [6:42:32<08:16,  3.91s/it]


Too many queries in the last time. Need to wait 268 seconds, until 20:16.


JSON Query to vitta.kv/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 96%|█████████▌| 2694/2813 [6:47:27<19:05,  9.63s/it]  


Too many queries in the last time. Need to wait 15 seconds, until 20:17.


 96%|█████████▌| 2707/2813 [6:48:31<09:02,  5.12s/it]


Too many queries in the last time. Need to wait 19 seconds, until 20:18.


 97%|█████████▋| 2727/2813 [6:50:08<04:09,  2.90s/it]


Too many queries in the last time. Need to wait 66 seconds, until 20:20.


 98%|█████████▊| 2760/2813 [6:53:35<03:26,  3.90s/it]


Too many queries in the last time. Need to wait 270 seconds, until 20:27.


JSON Query to kochetkovavv/feed/: ('Connection aborted.', ConnectionResetError(10054, 'Удаленный хост принудительно разорвал существующее подключение', None, 10054, None)) [retrying; skip with ^C]
 99%|█████████▉| 2781/2813 [6:59:36<02:29,  4.69s/it]  


Too many queries in the last time. Need to wait 16 seconds, until 20:29.


 99%|█████████▉| 2784/2813 [7:00:00<02:35,  5.38s/it]


Too many queries in the last time. Need to wait 19 seconds, until 20:29.


100%|█████████▉| 2801/2813 [7:01:14<00:35,  2.93s/it]


Too many queries in the last time. Need to wait 63 seconds, until 20:31.


100%|██████████| 2813/2813 [7:03:14<00:00,  9.03s/it]


**Delete the accounts that could not be parsed. Their share is 2%**

Удалим аккаунты, которые не удалось спарсить. Их доля составляет 2 %

In [19]:
df[df.is_business.isna()]

user_name is_business business_category is_private  \
8               ksenn2912         NaN               NaN        NaN   
15             rafinadova         NaN               NaN        NaN   
85               mosashop         NaN               NaN        NaN   
200           intervladiq         NaN               NaN        NaN   
285          polyak.fresh         NaN               NaN        NaN   
...                   ...         ...               ...        ...   
2501         primaoradiva         NaN               NaN        NaN   
2557  ministerstvo_uspeha         NaN               NaN        NaN   
2595             khomanov         NaN               NaN        NaN   
2618          workout.rus         NaN               NaN        NaN   
2660        andrewpetrov1         NaN               NaN        NaN   

      count_posts  count_followers  count_followees is_verified biography  \
8             NaN              NaN              NaN         NaN       NaN   
15            NaN              NaN              NaN         NaN       NaN   
85            NaN              NaN              NaN         NaN       NaN   
200           NaN              NaN              NaN         NaN       NaN   
285           NaN              NaN              NaN         NaN       NaN   
...           ...              ...              ...         ...       ...   
2501          NaN              NaN              NaN         NaN       NaN   
2557          NaN              NaN              NaN         NaN       NaN   
2595          NaN              NaN              NaN         NaN       NaN   
2618          NaN              NaN              NaN         NaN       NaN   
2660          NaN              NaN              NaN         NaN       NaN   

     full_name  count_igtv post_captions  lenth_title4post  likes4post  \
8          NaN         NaN           NaN               NaN         NaN   
15         NaN         NaN           NaN               NaN         NaN   
85         NaN         NaN           NaN               NaN         NaN   
200        NaN         NaN           NaN               NaN         NaN   
285        NaN         NaN           NaN               NaN         NaN   
...        ...         ...           ...               ...         ...   
2501       NaN         NaN           NaN               NaN         NaN   
2557       NaN         NaN           NaN               NaN         NaN   
2595       NaN         NaN           NaN               NaN         NaN   
2618       NaN         NaN           NaN               NaN         NaN   
2660       NaN         NaN           NaN               NaN         NaN   

      comms4post  hashtags4post  publications_period  
8            NaN            NaN                  NaN  
15           NaN            NaN                  NaN  
85           NaN            NaN                  NaN  
200          NaN            NaN                  NaN  
285          NaN            NaN                  NaN  
...          ...            ...                  ...  
2501         NaN            NaN                  NaN  
2557         NaN            NaN                  NaN  
2595         NaN            NaN                  NaN  
2618         NaN            NaN                  NaN  
2660         NaN            NaN                  NaN  

[66 rows x 17 columns]

In [22]:
idx = df[df.is_business.isna()].index
df = df.drop(idx).reset_index(drop=True)

In [23]:
df.shape[0]

2747

**As a result , we have 2747 rows**

По итогу парсинга имеем 2747 записи

In [11]:
df.reset_index(drop=True, inplace=True)

In [13]:
df.to_csv('data.csv', index=False)